src/exercise/02.js

```diff
@@ -6,11 +6,19 @@ import * as React from 'react'
 function Greeting({initialName = ''}) {
   // 🐨 initialize the state to the value from localStorage
   // 💰 window.localStorage.getItem('name') ?? initialName
-  const [name, setName] = React.useState(initialName)
-
+  const [name, setName] = React.useState(
+    () => window.localStorage.getItem('name') ?? initialName,
+  )
+   
+   
+   
   // 🐨 Here's where you'll use `React.useEffect`.
   // The callback should set the `name` in localStorage.
   // 💰 window.localStorage.setItem('name', name)
+  React.useEffect(() => {
+    window.localStorage.setItem('name', name)
+  })
```

Comment:
```suggestion
  }, [name])
```
il faut le faire seulement quand le `name` change 
sans passer le tableaux de dépendences ça va l'assigner sur chaque render dans le composant